src/map/script.cpp

```diff
@@ -17856,7 +17859,7 @@ BUILDIN_FUNC(setunitdata)
 					md->state.copy_master_mode = 0;
 				calc_status = true;
 				break;
-			case UMOB_DMGIMMUNE: md->ud.immune_attack = (bool)value > 0 ? 1 : 0; break;
+			case UMOB_DMGIMMUNE: md->ud.immune_attack = value > 0 ? 1 : 0; break;
```

Comment:
just remove that completely and make it `... = value > 0;`
goes for all below as well